community/spatial-index/src/main/java/org/neo4j/gis/spatial/index/curves/SpaceFillingCurve.java

```diff
@@ -143,6 +143,23 @@ private Long derivedValueFor( double[] coord, int level )
     {
         assertValidLevel( level );
         long[] normalizedValues = getNormalizedCoord( coord );
+        return derivedValueFor( normalizedValues, level );
+    }
+
+    /**
+     * Given a normalized coordinate in multiple dimensions, calculate its derived key for maxLevel
+     */
+    public Long derivedValueFor( long[] normalizedValues )
```

Comment:
This code comes from `Neo4j Spatial` where the value is placed into an explicit lucene index, not the GPBTree, and this method is required there. Moving forward instead of depending on Neo4j Spatial code, it will depend on Neo4j code, and this public API is part of the contract.

Reply:
Aah, I see. Can we document this to remind us not to "clean" this up and break `Neo4j Spatial` in the future?

Reply:
✅